cmd/scrape.go

```diff
@@ -0,0 +1,266 @@
+package cmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/gocolly/colly/v2"
+	"github.com/spf13/cobra"
+)
+
+var options             Options
+var url                 string
+
+type Article struct {
+	ID                  int             `json:"id"`
+    Title               string          `json:"title"`
+}
+
+type Site struct {
```

Comment:
As this struct can potentially have a non-trivial size and because the Site.Scrape method below mutates the value, I suggest that its methods below be converted to use pointer receivers (`func (s *Site) Scrape(...) {...}`).

While those are not safe for concurrency and may need channels in case we do crazy async stuff with its methods, I think it's better than polluting our stack too much by passing around entire copies of the stuff.